source/particle/generator/quadrature_points.cc

```diff
@@ -0,0 +1,130 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/generator/quadrature_points.h>
+
+#include <deal.II/fe/fe_values.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      template <int dim>
+      QuadraturePoints<dim>::QuadraturePoints()
+      {
+        starting_particle_index = 0;
+      }
+
+      template <int dim>
+      void
+      QuadraturePoints<dim>::generate_particles(std::multimap<types::LevelInd, Particle<dim> > &particles)
+      {
+        const QGauss<dim> quadrature_formula(this->get_parameters().stokes_velocity_degree + 1);
+
+        unsigned int n_particles_to_generate = 0, prefix_sum = 0;
+
+        FEValues<dim> fe_values(this->get_mapping(),
+                                this->get_fe(),
+                                quadrature_formula,
+                                update_values |
+                                update_quadrature_points |
+                                update_JxW_values);
+
+        typename parallel::distributed::Triangulation<dim>::active_cell_iterator
+        cell = this->get_triangulation().begin_active(),
+        endc = this->get_triangulation().end();
+
+        if (cell != endc)
+          {
+            fe_values.reinit(cell);
+            n_particles_to_generate = this->get_triangulation().n_locally_owned_active_cells() * fe_values.n_quadrature_points;
+          }
+
+        MPI_Scan(&n_particles_to_generate, &prefix_sum, 1, MPI_UNSIGNED, MPI_SUM, this->get_mpi_communicator());
```

Comment:
Could you change `MPI_UNSIGNED` to `ASPECT_TRACER_INDEX_MPI_TYPE`. Although no individual process will own more than 4 billion particles (although it is possible on machines with 512 GB RAM, I tested that once :smile:) the overall sum can be larger.